src/scheduler/server_info.c

```diff
@@ -1172,8 +1172,6 @@ free_server_info(server_info *sinfo)
 	if (sinfo->nodes_by_NASrank != NULL)
 		free(sinfo->nodes_by_NASrank);
 #endif
-
-	free(sinfo);
```

Comment:
I'm not sure if I like this, this will make it inconsistent with the other destructors work right? as an interface, one would expect free_server_info() to free the object itself as well. How about this instead: you add an additional argument to this function which says "also free the other reference objects like queues, nodes, resvs etc." so that you could do that inside this function itself?

Reply:
Well It depends what you call the "destructor" function.  Is it free_server() or free_server_info()?  The call to free sinfo was originally inside free_server_info(), but with it being moved to the top of free_server(), we can't.  We reference sinfo when freeing the queues, nodes, jobs, and reservations.  It has to be the last thing freed.  This is why I moved it.